### PR TITLE
Fix checksum memory issue and other bugs

### PR DESCRIPTION
**Pull request by ahamelers**
Closes https://github.com/CDL-Dryad/dryad-product-roadmap/issues/2430
Closes https://github.com/CDL-Dryad/dryad-product-roadmap/issues/3208
Closes https://github.com/CDL-Dryad/dryad-product-roadmap/issues/3181

The checksum issue is a very simple fix (it was getting the object from AWS that was overloading the memory and not working intermittently, switching to using a URL fixed the issue) but took me forever to test because of a simple error I made in testing. Sorry about that!

### CODE REVIEW

File: lib/stash/repo/submission_job.rb [hunk 1]
```diff
@@ -114,7 +114,7 @@ def copy_to_permanent_store(data_file)
         update = { storage_version_id: resource.id }
         if data_file.digest.nil?
           digest_type = 'sha-256'
```
Comment (ahamelers) [inline lib/stash/repo/submission_job.rb:116]: This is the change that fixes the checksum generation, if you want to make this change on the live server, @ryscher

**Comment by ryscher** [comment]
I made this change on the production server, BUT I don't think the current configuration works for files that are uploaded via URL. The current checksum statement relies on `staged_key`, using the default "staged" bucket. When files are uploaded via URL, they don't exist in this location. I think we can swap this out for the permanent location, but I didn't verify it.

File: lib/stash/repo/submission_job.rb [hunk 2]
```diff
@@ -103,7 +103,7 @@ def copy_to_permanent_store(data_file)
             logger.info("    #{data_file.url} ==> #{permanent_bucket}/#{permanent_key}")
             s3_perm.put_stream(s3_key: permanent_key, stream: f)
           end
-          @digest_input = data_file.url
```
Comment (ahamelers) [inline lib/stash/repo/submission_job.rb:106]: @ryscher The input for URL uploads is set here, and should not be an issue. However there is another issue which was causing checksum duplication—`digest_input` should be a local variable in both instances 🤦

**Comment by ryscher** [comment]
Great, thanks for tracking this down!

**Pull request merged**